Mage.Sets/src/mage/cards/a/AnzragsRampage.java

```diff
@@ -0,0 +1,183 @@
+package mage.cards.a;
+
+import java.util.UUID;
+
+import mage.abilities.Ability;
+import mage.abilities.DelayedTriggeredAbility;
+import mage.abilities.common.delayed.AtTheBeginOfNextEndStepDelayedTriggeredAbility;
+import mage.abilities.dynamicvalue.DynamicValue;
+import mage.abilities.effects.ContinuousEffect;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.DestroyAllEffect;
+import mage.abilities.effects.common.ReturnToHandTargetEffect;
+import mage.abilities.effects.common.continuous.GainAbilityTargetEffect;
+import mage.abilities.hint.ValueHint;
+import mage.abilities.keyword.HasteAbility;
+import mage.cards.*;
+import mage.constants.*;
+import mage.filter.StaticFilters;
+import mage.filter.common.FilterArtifactPermanent;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.events.ZoneChangeEvent;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+import mage.target.TargetCard;
+import mage.target.common.TargetCardInExile;
+import mage.target.targetpointer.FixedTarget;
+import mage.util.CardUtil;
+import mage.watchers.Watcher;
+
+/**
+ * @author Cguy7777
+ */
+public final class AnzragsRampage extends CardImpl {
+
+    private static final FilterArtifactPermanent filter = new FilterArtifactPermanent("artifacts you don't control");
+
+    static {
+        filter.add(TargetController.NOT_YOU.getControllerPredicate());
+    }
+
+    private static final ValueHint hint = new ValueHint(
+            "Artifacts put into graveyards from the battlefield this turn", AnzragsRampageValue.instance);
+
+    public AnzragsRampage(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.SORCERY}, "{3}{R}{R}");
+
+        // Destroy all artifacts you don't control, then exile the top X cards of your library,
+        // where X is the number of artifacts that were put into graveyards from the battlefield this turn.
+        // You may put a creature card exiled this way onto the battlefield. It gains haste.
+        // Return it to your hand at the beginning of the next end step.
+        this.getSpellAbility().addEffect(new DestroyAllEffect(filter));
+        this.getSpellAbility().addEffect(new AnzragsRampageEffect().concatBy(", then"));
+        this.getSpellAbility().addWatcher(new AnzragsRampageWatcher());
+        this.getSpellAbility().addHint(hint);
+    }
+
+    private AnzragsRampage(final AnzragsRampage card) {
+        super(card);
+    }
+
+    @Override
+    public AnzragsRampage copy() {
+        return new AnzragsRampage(this);
+    }
+}
+
+// CardsPutIntoGraveyardWatcher does not count tokens so custom watcher is needed.
```

Comment:
No need ref class comments for simple code like watcher.

Reply:
Personally I think that is a useful comment. It's not just referencing, but explaining the reasoning behind them being separate (can protect against ill-advised refactors that miss the distinction)